File: app/src/main/java/com/zhuazhu/frame/di/component/AppComponent.java
package com.zhuazhu.frame.di.component;

import com.zhuazhu.frame.data.HttpHelper;
import com.zhuazhu.frame.di.module.MainModuleBase;
import com.zhuazhu.frame.di.module.NetModule;
import com.zhuazhu.frame.di.module.RecyclerModuleBase;

import javax.inject.Singleton;

import dagger.Component;
import mejust.frame.di.module.AppModule;

/**
 * @author : Beaven
 * @date : 2017-12-20 11:41
 */
@Singleton
@Component(modules = {NetModule.class, AppModule.class})
public interface AppComponent {

    HttpHelper getHttpHelper();

    MainComponent mainComponent(MainModuleBase mainModule);

    RecyclerComponent recyclerComponent(RecyclerModuleBase module);
}

File: app/src/main/java/com/zhuazhu/frame/di/component/AppComponent.kt
package com.zhuazhu.frame.di.component

import com.zhuazhu.frame.di.module.AppModule
import com.zhuazhu.frame.di.module.HttpModule
import dagger.BindsInstance
import dagger.Component
import mejust.frame.di.component.FrameComponent
import javax.inject.Singleton

/**
 * @author wangpeifeng
 * @date 2018/06/04 15:38
 */
@Singleton
@Component(modules = [(AppModule::class)])
interface AppComponent {

    fun httpComponent(httpModule: HttpModule): HttpComponent

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun frameComponent(frameComponent: FrameComponent): Builder

        fun build(): AppComponent
    }
}